packages/jaeger-ui/src/components/SearchTracePage/SearchForm.jsx

```diff
@@ -257,338 +257,338 @@ export function submitForm(fields, searchTraces) {
   });
 }
 
-export class SearchFormImpl extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      formData: {
-        service: this.props.initialValues?.service,
-        operation: this.props.initialValues?.operation,
-        tags: this.props.initialValues?.tags,
-        lookback: this.props.initialValues?.lookback,
-        startDate: this.props.initialValues?.startDate,
-        startDateTime: this.props.initialValues?.startDateTime,
-        endDate: this.props.initialValues?.endDate,
-        endDateTime: this.props.initialValues?.endDateTime,
-        minDuration: this.props.initialValues?.minDuration,
-        maxDuration: this.props.initialValues?.maxDuration,
-        resultsLimit: this.props.initialValues?.resultsLimit,
-      },
-    };
-  }
+export function SearchFormImpl(props) {
+  const {
+    invalid,
+    searchMaxLookback,
+    services,
+    submitting,
+    initialValues,
+    changeServiceHandler,
+    submitFormHandler,
+    handleChange: propHandleChange,
+  } = props;
+
+  const [formData, setFormData] = React.useState({
+    service: initialValues?.service,
+    operation: initialValues?.operation,
+    tags: initialValues?.tags,
+    lookback: initialValues?.lookback,
+    startDate: initialValues?.startDate,
+    startDateTime: initialValues?.startDateTime,
+    endDate: initialValues?.endDate,
+    endDateTime: initialValues?.endDateTime,
+    minDuration: initialValues?.minDuration,
+    maxDuration: initialValues?.maxDuration,
+    resultsLimit: initialValues?.resultsLimit,
+  });
 
-  handleChange = fieldData => {
-    this.setState(prevState => ({
-      formData: {
-        ...prevState.formData,
-        ...fieldData,
-      },
+  const internalHandleChange = fieldData => {
+    setFormData(prevState => ({
+      ...prevState,
+      ...fieldData,
     }));
     if (fieldData.service) {
-      this.props.changeServiceHandler(fieldData.service);
-      this.setState(prevState => ({
-        formData: {
-          ...prevState.formData,
-          operation: DEFAULT_OPERATION,
-        },
+      changeServiceHandler(fieldData.service);
+      setFormData(prevState => ({
+        ...prevState,
+        operation: DEFAULT_OPERATION,
       }));
     }
   };
 
-  handleSubmit = e => {
+  const handleChange = propHandleChange || internalHandleChange;
```

Comment:
the search form exists in only one place in the UI. What is the use case for this either/or behavior? Can we simplify to only one behavior?